earth_enterprise/src/fusion/autoingest/StorageManager.h

```diff
@@ -173,14 +179,14 @@ template<class AssetType>
 inline void
 StorageManager<AssetType>::AddExisting(const AssetKey & key, const PointerType & value) {
   std::lock_guard<std::recursive_mutex> lock(storageMutex);
-  cache.Add(key, value);
+  cache.Add(key, value, DetermineIfPrune());
 }
 
 template<class AssetType>
 inline void
 StorageManager<AssetType>::NoLongerNeeded(const AssetKey & key, bool prune) {
   std::lock_guard<std::recursive_mutex> lock(storageMutex);
-  cache.Remove(key, prune);
+  cache.Remove(key, (DetermineIfPrune() && prune));
```

Comment:
It looks like the `prune` parameter is never used in the `NoLongerNeeded` function. I'd suggest removing it and just passing `DetermineIfPrune()` to `cache.Remove`.

Reply:
This is needed for various unit tests in cases where we specifically do not want to prune the cache. If false is passed to NoLongerNeeded, then the cache will not be pruned despite the result of DetermineIfPrune.

Reply:
If it's needed for the unit tests I'm fine with leaving it in.

Reply:
There are various places in StorageManager.h where it is specifically stated that the cache should not be pruned after a remove, but to wait until after an item is added. With this in mind, I think we should leave this argument as is.